eclipse-collections/src/main/java/org/eclipse/collections/impl/list/mutable/FastList.java

```diff
@@ -167,6 +167,11 @@ public FastList(Collection<? extends T> source)
 
     public static <E> FastList<E> newList(int initialCapacity)
     {
+        if (initialCapacity < 0)
```

Comment:
This is an API breaking changing on implementation. This check needs to move up in the factory class and for next major version we should move it down to FastList.